### ISSUE TITLE
fCal (orientation problems after calibration)

### ISSUE THREAD

**Issue by joaolgfonseca** [PlusToolkit/PlusLib#286]
To whom it may concern,

First of all congratulations for PLUS Toolkit and for sharing it.

Currently, I am trying to use PLUS Toolkit to grab information from an ultrasound machine (using the ephiphan video to get the image from the display monitor) and an NDI Aurora electromagnetic sensor. Besides, I am using the fCAL software to perform calibration between these two equipments (using a convex probe and a 6-DOF sensor).

The setup is composed of a 5-DOF needle (used as a stylus), the fCal_3.1 phantom (3D printed) with a 6 DOF reference attached and a tank with ethanol diluted in a distilled water mimicking the speed of sound. I can perform all the steps of the fCAL and obtain all the transformations.  However, the spatial calibration is not correct. I continuously obtain errors higher than 2mm (sometimes 10-30mm) and my biggest concern is that some movements are not correct. Mainly the representation of the rotation movements on the fCAL software after calibration.

I tested your optimization algorithms for N-wires method, but I am still having the same orientation errors. I followed your recommendations in documentation to solve other problems that I had to adapt examples to the fCal_3.1 phantom and to the US Convex probe, but the answer to this problem I can not find.

In attach, I am sending a video where you can examine these problems. Do you have some ideas what can be happening?

Best regards!

[VID_20171205_095542_edit.zip](https://github.com/PlusToolkit/PlusLib/files/1548233/VID_20171205_095542_edit.zip)

**Comment by Sunderlandkyl** [comment]
@lassoan

**Comment by adamrankin** [comment]
A very basic question, but could you confirm that the wiring of the phantom matches the configuration file? Could you take a few pictures of your fCal_3.1 and your calibration configuration file's `<PhantomDefinition>` section.

**Comment by adamrankin** [comment]
Additionally, what is your stylus error, and your phantom error?

**Comment by joaolgfonseca** [comment]
Thank you very much for your concern.

We extensively examine the equipment and we found that the 6-DOF sensor (bought recently) attached to the probe is not working properly. The rotations, obtained from some axis, are incorrectly computed by NDI Aurora. We think this is the cause of the orientation problems. In addition, we tested the alternative calibration method (pointer-based calibration) using a different 6-DOF sensor and the calibration was completed with success (using a pointer with 5-DOF sensor).

I just have one 6-DOF sensor available at the moment. Can I use two 5-DOF as stylus and reference without compromise the calibration too much?

Furthermore, I will send you some information about the setup.

**Errors:** 
Stylus: 0.870298
Phantom: 2.07016

[images.zip](https://github.com/PlusToolkit/PlusLib/files/1559657/images.zip)

Thank you very much!
